Coupled-Cluster/RHF/helper_cchbar.py

```diff
@@ -0,0 +1,254 @@
+
+    def __init__(self, ccsd, memory=2):
+
+        # Start of the cchbar class
+        time_init = time.time()
+
+        self.MO = ccsd.MO
+        self.ndocc = ccsd.ndocc
+        self.nmo = ccsd.nmo
+        self.nocc = ccsd.ndocc
+        self.nvirt = ccsd.nmo - ccsd.nocc
+
+        self.slice_o = slice(0, self.nocc)
+        self.slice_v = slice(self.nocc, self.nmo)
+        self.slice_a = slice(0, self.nmo)
+        self.slice_dict = {'o' : self.slice_o, 'v' : self.slice_v,
+                           'a' : self.slice_a}
+
+
+        self.F = ccsd.F
+        self.Dia = ccsd.Dia
+        self.Dijab = ccsd.Dijab
+        self.t1 = ccsd.t1
+        self.t2 = ccsd.t2
+
+        print('\nBuilding HBAR components ...')
+
+        self.build_Loovv() 
+        self.build_Looov() 
+        self.build_Lvovv() 
+
+        self.build_Hov()
+        self.build_Hoo()
+        self.build_Hvv()
+        self.build_Hoooo()
+        self.build_Hvvvv()
+        self.build_Hvovv()
+        self.build_Hooov()
+        self.build_Hovvo()
+        self.build_Hovov()
+        self.build_Hvvvo()
+        self.build_Hovoo()
+
+        print('\n..HBAR Build completed !!')
+
+    # occ orbitals i, j, k, l, m, n
+    # virt orbitals a, b, c, d, e, f
+    # all oribitals p, q, r, s, t, u, v
+
+    def get_MO(self, string):
+        if len(string) != 4:
+            psi4.core.clean()
+            raise Exception('get_MO: string %s must have 4 elements.' % string)
+        return self.MO[self.slice_dict[string[0]], self.slice_dict[string[1]],
+                     self.slice_dict[string[2]], self.slice_dict[string[3]]]
+
+    def get_F(self, string):
+        if len(string) != 2:
+            psi4.core.clean()
+            raise Exception('get_F: string %s must have 2 elements.' % string)
+        return self.F[self.slice_dict[string[0]], self.slice_dict[string[1]]]
+
+    def build_Loovv(self):
+        tmp = self.get_MO('oovv').copy()
+        self.Loovv = 2.0 * tmp - tmp.swapaxes(2,3)
+        return self.Loovv
+
+    def build_Looov(self):
+        tmp = self.get_MO('ooov').copy()
+        self.Looov = 2.0 * tmp - tmp.swapaxes(0,1)
+        return self.Looov
+
+    def build_Lvovv(self):
+        tmp = self.get_MO('vovv').copy()
+        self.Lvovv = 2.0 * tmp - tmp.swapaxes(2,3)
+        return self.Lvovv
+
+    def build_tau(self):
+        self.ttau = self.t2.copy()
+        tmp = np.einsum('ia,jb->ijab', self.t1, self.t1)
+        self.ttau += tmp
+        return self.ttau
+
+    # F and W are the one and two body intermediates which appear in the CCSD
+    # T1 and T2 equations. Please refer to helper_ccenergy file for more details.
+    # Spin orbitals expression of all Hbar components are written in einstein notation in the 
+    # comments below. Ofcourse we are constructing <p * alpha | Hbar |q * alpha> (or beta) 
+    # and <p * alpha * q * beta | Hbar | r * alpha * s * beta> components here.
+
+    # <m|Hbar|e> = F_me = f_me + t_nf <mn||ef>
```

Comment:
Can you move these strings to function doc strings so that they can be picked up by the Python interpreter?

Reply:
Done!